authority/tls.go

```diff
@@ -362,6 +364,16 @@ func (a *Authority) Revoke(ctx context.Context, revokeOpts *RevokeOptions) error
 		p   provisioner.Interface
 		err error
 	)
+
+	// Attempt to get the certificate expiry using the serial number.
+	cert, err := a.db.GetCertificate(revokeOpts.Serial)
+
+	// Revocation of a certificate not in the database may be requested, so fill in the expiry only
+	// if we can
+	if err == nil {
+		rci.ExpiresAt = cert.NotAfter
+	}
```

Comment:
Where is the info for the mTLS auth stored? Its been a while since looking at this, so it may be quicker for you to point me at the NotAfter that can be used in mTLS calls... :)

Reply:
RevokeOptions has a certificate if mTLS is used. See 
https://github.com/smallstep/certificates/blob/5cb23c6029fe775a303b5613f6d4374c4df2550c/authority/tls.go#L363-L372

A way to test this is to generate a certificate and then revoke it using: 
```console
step ca revoke --cert foo.crt --key foo.key
```

Instead of using just the serial.

Reply:
Basically, this lookup is only necessary if Crt is nil.